src/items/functions.md

```diff
@@ -214,12 +214,51 @@ extern "C" fn new_i32() -> i32 { 0 }
 let fptr: extern "C" fn() -> i32 = new_i32;
 ```
 
-Functions with an ABI that differs from `"Rust"` do not support unwinding in the
-exact same way that Rust does. Therefore, unwinding past the end of functions
-with such ABIs causes the process to abort.
-
-> **Note**: The LLVM backend of the `rustc` implementation
-aborts the process by executing an illegal instruction.
+### Unwinding
+
+Most ABI strings come in two variants, one with an `-unwind` suffix and one without.
+The `Rust` ABI always permits unwinding, so there is no `Rust-unwind` ABI.
+
+The choice of ABI, together with the [panic mode][panic-modes], determines the
+behavior when unwinding out of a function.
+
+The table below indicates the behavior of an unwinding operation reaching each
+type of ABI boundary (function declaration or definition using the
+corresponding ABI string). Note that the Rust runtime is not affected by, and
+cannot have an effect on, any unwinding that occurs entirely within another
+language's runtime, that is, unwinds that are thrown and caught without
+reaching a Rust ABI boundary.
+
+The "unwinding" ABI category refers to `"Rust"` (the implicit ABI of Rust
+functions not marked `extern`), `"C-unwind"`, and any other ABI with `-unwind`
+in its name. The "non-unwinding" ABI category refers to all other ABI strings,
+including `"C"` and `"stdcall"`.
+
+Native unwinding is defined per-target. On targets that support throwing and
+catching C++ exceptions, it refers to the mechanism used to implement this
+feature. Some platforms implement a form of unwinding referred to as ["forced
+unwinding"][forced-unwinding]; `longjmp` on Windows and `pthread_exit` in
+`glibc` are implemented this way. Forced unwinding is explicitly excluded
+from the "Native unwind" column in the table.
+
+| panic runtime  | ABI           | `panic`-unwind                        | Native unwind (unforced) |
+| -------------- | ------------  | ------------------------------------- | -----------------------  |
+| `panic=unwind` | unwinding     | unwind                                | unwind                   |
+| `panic=unwind` | non-unwinding | abort[^1]                             | [undefined behavior]     |
+| `panic=abort`  | unwinding     | `panic!` aborts (no unwinding occurs) | abort                    |
+| `panic=abort`  | non-unwinding | `panic!` aborts (no unwinding occurs) | [undefined behavior]     |
+
+[^1]: In this case, either no destructors (`Drop` calls) will run, or all
+      destructors up until the `extern "C"` boundary (or other such "no unwind"
+      boundary) will run.
```

Comment:
This seems important enough to not just be in a footnote.

Also isn't `panic.md` a better place for this? (Not sure, just wondering.)

Reply:
I thought about whether this belongs in the new panic documentation, but this section is more about the ABI strings, which are part of the function definition & declaration, so I left it here. I added cross-links, though.

I'm not sure whether the footnote content is actually important enough not to be a footnote. The important part for most users, I think, is simply that an abort is guaranteed. If you need to know whether a particular destructor will be run before termination, there's a footnote for that.

Reply:
"Which of your code actually is guaranteed to rune" seems like an extremely important guarantee to me. That can't just be a footnote.

Though this also seems like the wrong file for that. Maybe [here](https://doc.rust-lang.org/reference/destructors.html) is a better place?

Reply:
I've changed it from a footnote to a note.

This is the right place for the note because it's clarifying a specific cell in the table here. The `destructors` documentation probably _should_ have some references to the new `panic` documentation, though, shouldn't it?

Reply:
...oh, I guess I must have already added a link from `destructors.md` to `panic.md`.

I'll mention the thing about `abort` not running destructors in the "not running destructors" section.

Reply:
Added